pkg/util/util.go

```diff
@@ -57,5 +58,22 @@ func NamespaceOpenShiftObject(componentName string, applicationName string) (str
 	}
 
 	// Return the hyphenated namespaced name
-	return fmt.Sprintf("%s-%s", componentName, applicationName), nil
+	return fmt.Sprintf("%s-%s", strings.Replace(componentName, "/", "-", -1), applicationName), nil
+}
+
+// ExtractComponentType returns only component type part from passed component type(default unqualified, fully qualified, versioned, etc...and their combinations) for use as component name
+func ExtractComponentType(namespacedVersionedComponentType string) string {
+	/*
+		Possible types of parameters:
+		1. "myproject/python:3.5" -- Return python
+		2. "python:3.5" -- Return python
+		3. nodejs -- Return nodejs
+	*/
```

Comment:
these comments should be added above (where you describe ExtractComponentType

Reply:
Ack